lib/network_manager.go

```diff
@@ -77,18 +98,25 @@ type NetworkManager struct {
 	exitGroup  sync.WaitGroup
 }
 
-func NewNetworkManager(params *DeSoParams, cmgr *ConnectionManager, rnManager *RemoteNodeManager,
+func NewNetworkManager(params *DeSoParams, srv *Server, bc *Blockchain, cmgr *ConnectionManager,
 	blsKeystore *BLSKeystore, addrMgr *addrmgr.AddrManager, connectIps []string,
 	targetNonValidatorOutboundRemoteNodes uint32, targetNonValidatorInboundRemoteNodes uint32,
-	limitOneInboundConnectionPerIP bool) *NetworkManager {
+	limitOneInboundConnectionPerIP bool, minTxFeeRateNanosPerKB uint64, nodeServices ServiceFlag) *NetworkManager {
 
 	return &NetworkManager{
 		params:                                params,
+		srv:                                   srv,
+		bc:                                    bc,
 		cmgr:                                  cmgr,
-		blsKeystore:                           blsKeystore,
-		handshake:                             NewHandshakeController(rnManager),
-		rnManager:                             rnManager,
+		keystore:                              blsKeystore,
 		AddrMgr:                               addrMgr,
+		minTxFeeRateNanosPerKB:                minTxFeeRateNanosPerKB,
+		nodeServices:                          nodeServices,
+		AllRemoteNodes:                        collections.NewConcurrentMap[RemoteNodeId, *RemoteNode](),
+		ValidatorIndex:                        collections.NewConcurrentMap[bls.SerializedPublicKey, *RemoteNode](),
+		NonValidatorOutboundIndex:             collections.NewConcurrentMap[RemoteNodeId, *RemoteNode](),
+		NonValidatorInboundIndex:              collections.NewConcurrentMap[RemoteNodeId, *RemoteNode](),
+		usedNonces:                            lru.NewCache(1000),
```

Comment:
I assume this is to make it so that you can recognize when a validator is reusing a nonce? If yes I would bump it to a higher number like 100k since we could have a lot of validator connections coming in and a lot of nonces to track.